exonum/src/encoding/segments.rs

```diff
@@ -311,10 +311,20 @@ impl<'a> SegmentField<'a> for &'a [u8] {
     }
 }
 
-/// Implement field helper for all array of POD types
-/// it writes POD type as bytearray in place.
+/// Implements a field helper for an array of POD type. This macro enables to
+/// convert POD type data into a byte array.
```

Comment:
I have a slight issue with this description. As far as I understand the macro allows to use fields with type `&[T]` (where `T` is the argument of the macro) in Exonum persistence mechanisms (i.e., `transactions!` and `encoding_struct!` macros). `T` needs to be a POD (plain old data) type for the conversion to work.

Also, `as *const Hash` in the implementation looks like an error: https://github.com/exonum/exonum/blob/cc31cd4579c00675de611135c6264c288549d513/exonum/src/encoding/segments.rs#L331-L336